subsys/bluetooth/host/mesh/cfg_cli.c

```diff
@@ -1016,14 +1016,24 @@ int bt_mesh_cfg_hb_pub_get(u16_t net_idx, u16_t addr, u16_t *pub_dst,
 	cli->op_param = &param;
 	cli->op_pending = OP_HEARTBEAT_PUB_STATUS;
 
-	err = k_sem_take(&cli->op_sync, MSG_TIMEOUT);
+	err = k_sem_take(&cli->op_sync, msg_timeout);
 
 	cli->op_pending = 0;
 	cli->op_param = NULL;
 
 	return err;
 }
 
+s32_t bt_mesh_cfg_cli_timeout_get(void)
+{
+	return msg_timeout;
+}
+
+void bt_mesh_cfg_cli_timeout_set(s32_t timeout)
```

Comment:
It is not very clear why you didn't went with timeout per call instead of a global one that affects every function? This type of API is a bit inconvenient if there are multiple threads using since each one may change the timeout the threads will always have to get the current value first to check if it still what it wants.

Reply:
I added a comment to the commit message clarifying its usage.